### PR TITLE
Story3: Typing words

### PR DESCRIPTION
**Pull request by melody-sylvestre**
Description:

As a: player
I want: to be able to type words into the game and see if they are correct
So that: I can play the game and see how I’m doing

Requirements:

    The user types a word and then presses space, which locks in the word
    If the word was wrong, it becomes red in the source text
    If correct, it becomes green in the source text
    As the user types a space, the textbox they are typing in becomes blank, ready for them to type the next word

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -8,10 +15,10 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
         //the line below replaces any accented character by the matching non-accented character
         //e.g. "é" becomes "e"
         textOutput = textOutput.normalize("NFD").replace(/\p{Diacritic}/gu, "")
-        wordsArray =  textOutput.split(' ')
+        wordsArray = textOutput.split(' ')
```
Comment (mporam) [inline script.js]: what is `wordsArray`? you havent declared it as a let or const and you appear to immediately discard it so could could just assign the split on the line below

File: script.js [hunk 2]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
     })
     stringForHtml += '</p>'
     textToTypeContainer.innerHTML += stringForHtml
+    let upcomingWord = document.getElementById(0)
+    upcomingWord.style.color = ("#cc7a00")
```
Comment (mporam) [inline script.js]: dont need brackets, and this isnt very readable, I would suggest putting the colours into descriptive constants at the top, that will make this make more sense

File: script.js [hunk 3]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+    upcomingWordLength = upcomingWord.clientWidth
+    totalLengthOfWords += upcomingWordLength
 })
 
 
 
+textInput.addEventListener('keyup', event => {
+    if (event.code === 'Space') {
+        let textValue = textInput.value
+        let textValueWithoutSpace = textValue.slice(0, textValue.length - 1);
```
Comment (mporam) [inline script.js]: inconsistent use of `;`

File: script.js [hunk 4]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+        wordIndexCount++
+        wordIsValid = (textValueWithoutSpace === referenceTextArray[wordIndexCount])
+
+        let wordJustFinished = document.getElementById(wordIndexCount)
+        if (wordIsValid) {
+            numberOfValidWords++ // for later use
+            textInput.value = ''
```
Comment (mporam) [inline script.js]: this line and line 51 dont need to be in the if/else as it happens in both scenarios

File: script.js [hunk 5]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+            wordJustFinished.style.color = ("green")
```
Comment (mporam) [inline script.js]: no brackets

File: script.js [hunk 6]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+        } else {
+            textInput.value = ''
+            wordJustFinished.style.color = ("red")
```
Comment (mporam) [inline script.js]: no brackets

File: script.js [hunk 7]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
     })
     stringForHtml += '</p>'
     textToTypeContainer.innerHTML += stringForHtml
+    let upcomingWord = document.getElementById(0)
```
Comment (mporam) [inline script.js]: i know technically this was in a previous story but unfortunately this is invalid HTML. As you will see on the notes in this page:
https://www.w3schools.com/html/html_id.asp
HTML `id` attributes cannot start with a number and must contain at least one non-numerical character. so you need to make your ids something like: `word-0`, `word-1` then concatenate `word-` to your `getElementById`

File: script.js [hunk 8]
```diff
@@ -20,8 +27,51 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
...
+            let lineHeight = upcomingWord.clientHeight
+            totalLengthOfWords = 0
+            totalLengthOfWords += upcomingWordLength
+            textToTypeContainer.scroll({
+                top: scrollPixels += lineHeight,
+                behavior: 'smooth'
+            })
+        
+        }
+        upcomingWord.style.color = ("#cc7a00")
```
Comment (mporam) [inline script.js]: again, needs a constant

File: script.js (hunk 5 above)
Comment (mporam) [reply script.js]: also probably use a constant here incase you need to repeat it later

File: script.js (hunk 6 above)
Comment (mporam) [reply script.js]: also probably use a constant here incase you need to repeat it later

**Review by mporam — CHANGES_REQUESTED** [review]

File: script.js (hunk 1 above)
Comment (melody-sylvestre) [reply script.js]: done

File: script.js (hunk 3 above)
Comment (melody-sylvestre) [reply script.js]: I removed it

File: script.js (hunk 4 above)
Comment (melody-sylvestre) [reply script.js]: I moved line 48 outside if the if/else and removed line 51

File: script.js (hunk 2 above)
Comment (melody-sylvestre) [reply script.js]: Done: I declared constants for each word colour (see lines 10 to 12) and used them each time the colour of a word is changed.

File: script.js (hunk 5 above)
Comment (melody-sylvestre) [reply script.js]: Done

File: script.js (hunk 6 above)
Comment (melody-sylvestre) [reply script.js]: Done

File: script.js (hunk 8 above)
Comment (melody-sylvestre) [reply script.js]: Done

File: script.js (hunk 7 above)
Comment (melody-sylvestre) [reply script.js]: Fixed it, all words are now accessible through HTML ids like `word-0`, `word-1` etc.

**Review by mporam — APPROVED** [review]

**Pull request merged**